File: src/db/test_data/upvote.js
const casual = require('casual');
const times = require('../utils/times');

const upvote = (users_limit, contents_limit) => {
    const score = casual.double((from = 0), (to = 1000));
    return {
        voter_id: casual.integer((from = 1), (to = users_limit)),
        voted_id: casual.integer((from = 1), (to = contents_limit)),
        country_code: 'JP',
        score: score,
        pure_score: score,
        last_payout_score: score,
        last_payout_pure_score: score,
        author_score: score * 0.9,
        author_pure_score: score * 0.9,
        last_payout_author_score: score * 0.9,
        last_payout_author_pure_score: score * 0.9,
        voter_score: score * 0.1,
        voter_pure_score: score * 0.1,
        last_payout_voter_score: score * 0.1,
        last_payout_voter_pure_score: score * 0.1,
        max_accepted_payout: '',
        hasPendingPayout: false,
        deadline_cashout_at: new Date(),
        last_payout_at: new Date(),
        last_score_at: new Date(),
        isPrivate: false,
        valid: true,
        isCheering: true,
        isBetterOpinion: false,
        permission: true,
        created_at: new Date(),
        updated_at: new Date(),
    };
};

async function upvotes(limit = 30, users_number = 30, contents_number = 30) {
    let upvotes_array = [];
    await times(limit)(() => {
        upvotes_array.push(upvote(users_number, contents_number));
    });
    return upvotes_array;
}

module.exports = upvotes;

File: src/python-scripts/models/upvote.py

class Upvotes(object):
    def __init__(self, JSON=None):
        if JSON == None:
            self.items = []
            return
        self.items = [Upvote(upvote) for upvote in JSON if upvote is not None]

    def toSortArrayWithContent(self):
        return { val.VotedId : [ item.VoterId for item in self.getsByVotedId(val.VotedId) ] for val in self.items }

    def getsByVotedId(self, id):
        return [item for item in self.items if item.VotedId == id]

class Upvote(object):
    def __init__(self, JSON=None):
        if JSON == None:
            return
        self.id = JSON.get("id")
        self.VoterId = JSON.get("VoterId")
        self.VotedId = JSON.get("VotedId")
        self.score = JSON.get("score")
        self.isPrivate = JSON.get("isPrivate")
        self.valid = JSON.get("valid")
        self.permission = JSON.get("permission")
